Source/WebCore/html/track/WebVTTParser.cpp

```diff
@@ -648,6 +648,20 @@ static WebVTTNodeType tokenToNodeType(WebVTTToken& token)
     return WebVTTNodeTypeNone;
 }
 
+static WTF::String serializeTimestamp(double timestamp)
+{
+    uint64_t total = timestamp * 1000;
+    unsigned milliseconds = total % 1000;
+    total /= 1000;
+    unsigned seconds = total % 60;
+    total /= 60;
+    unsigned minutes = total % 60;
+    unsigned hours = total / 60;
+    char buffer[13];
+    snprintf(buffer, sizeof(buffer), "%02u:%02u:%02u.%03u", hours, minutes, seconds, milliseconds);
+    return String::fromUTF8(buffer);
```

Comment:
It would be better to use `StringBuilder` than `snprintf`. Maybe something like this:

```
    StringBuilder builder;

    uint64_t total = timestamp * 1000;
    unsigned milliseconds = total % 1000;
    total /= 1000;
    unsigned seconds = total % 60;
    total /= 60;
    unsigned minutes = total % 60;
    unsigned hours = total / 60;

    appendNumber<2>(builder, hours);
    builder.append(':');
    appendNumber<2>(builder, minutes);
    builder.append(':');
    appendNumber<2>(builder, seconds);
    builder.append('.');
    appendNumber<3>(builder, milliseconds);

    return builder.toString().impl();
```

Reply:
@eric-carlson PTAL.